diff --git a/CMakeLists.txt b/CMakeLists.txt
index 35827be..bb4eb4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,9 +42,9 @@ FetchContent_MakeAvailable(
 # Enable JUCE. Do not use find_package to prevent from mix up with one globally installed.
 add_subdirectory(lib/JUCE)
 
-enable_testing()
-configure_file(${CMAKE_SOURCE_DIR}/CTestCustom.cmake ${CMAKE_BINARY_DIR})
+#enable_testing()
+#configure_file(${CMAKE_CURRENT_LIST_DIR}/CTestCustom.cmake ${CMAKE_BINARY_DIR})
 
 add_subdirectory(src)
-add_subdirectory(tests)
-add_subdirectory(benchmark)
+#add_subdirectory(tests)
+#add_subdirectory(benchmark)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a907dcc..5a5cc83 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -65,43 +65,43 @@ target_compile_definitions(Os251
         )
 
 set(BINARY_SOURCES
-    ${CMAKE_SOURCE_DIR}/assets/Recursive-Regular.ttf
-
-    ${CMAKE_SOURCE_DIR}/assets/images/left.png
-    ${CMAKE_SOURCE_DIR}/assets/images/left_over.png
-    ${CMAKE_SOURCE_DIR}/assets/images/left_down.png
-
-    ${CMAKE_SOURCE_DIR}/assets/images/right.png
-    ${CMAKE_SOURCE_DIR}/assets/images/right_over.png
-    ${CMAKE_SOURCE_DIR}/assets/images/right_down.png
-
-    ${CMAKE_SOURCE_DIR}/assets/images/reload.png
-    ${CMAKE_SOURCE_DIR}/assets/images/reload_over.png
-    ${CMAKE_SOURCE_DIR}/assets/images/reload_down.png
-
-    ${CMAKE_SOURCE_DIR}/assets/presets/Default.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Bass/Bass0.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Bass/Bass1.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Bass/Bass2.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Bass/Bass3.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/FX/FX0.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/FX/FX1.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/FX/FX2.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/FX/FX3.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Key/Key0.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Key/Key1.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Key/Key2.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Key/Key3.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Pad/Pad0.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Pad/Pad1.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Pad/Pad2.oapreset
-    ${CMAKE_SOURCE_DIR}/assets/presets/Factory/Pad/Pad3.oapreset
+    ../assets/Recursive-Regular.ttf
+
+    ../assets/images/left.png
+    ../assets/images/left_over.png
+    ../assets/images/left_down.png
+
+    ../assets/images/right.png
+    ../assets/images/right_over.png
+    ../assets/images/right_down.png
+
+    ../assets/images/reload.png
+    ../assets/images/reload_over.png
+    ../assets/images/reload_down.png
+
+    ../assets/presets/Default.oapreset
+    ../assets/presets/Factory/Bass/Bass0.oapreset
+    ../assets/presets/Factory/Bass/Bass1.oapreset
+    ../assets/presets/Factory/Bass/Bass2.oapreset
+    ../assets/presets/Factory/Bass/Bass3.oapreset
+    ../assets/presets/Factory/FX/FX0.oapreset
+    ../assets/presets/Factory/FX/FX1.oapreset
+    ../assets/presets/Factory/FX/FX2.oapreset
+    ../assets/presets/Factory/FX/FX3.oapreset
+    ../assets/presets/Factory/Key/Key0.oapreset
+    ../assets/presets/Factory/Key/Key1.oapreset
+    ../assets/presets/Factory/Key/Key2.oapreset
+    ../assets/presets/Factory/Key/Key3.oapreset
+    ../assets/presets/Factory/Pad/Pad0.oapreset
+    ../assets/presets/Factory/Pad/Pad1.oapreset
+    ../assets/presets/Factory/Pad/Pad2.oapreset
+    ../assets/presets/Factory/Pad/Pad3.oapreset
 )
 
 if (UNIX)
-    list(APPEND BINARY_SOURCES ${CMAKE_SOURCE_DIR}/assets/Recursive-Regular.ttf)
+    list(APPEND BINARY_SOURCES ../assets/Recursive-Regular.ttf)
 else()
-    list(APPEND BINARY_SOURCES ${CMAKE_SOURCE_DIR}/assets/Os251Font125Percent.ttf)
+    list(APPEND BINARY_SOURCES ../assets/Os251Font125Percent.ttf)
 endif()
 
 juce_add_binary_data(Os251Binaries
@@ -144,4 +144,49 @@ set_target_properties(Os251Binaries PROPERTIES
         POSITION_INDEPENDENT_CODE TRUE
         )
 
+# must be added before juce_generate_juce_header()
+juce_add_module(${AAP_JUCE_DIR}/aap-modules/aap_audio_processors)
+
 juce_generate_juce_header(Os251)
+
+# begin JUCEAAP specific -->
+# It is not beautiful, but there does not seem to be any way to add custom include directory for modules.
+include_directories(
+        "${AAP_DIR}/include"
+)
+
+target_link_libraries(Os251
+        PRIVATE
+        aap_audio_processors
+)
+
+if (ANDROID)
+
+    # dependencies
+    find_library(log "log")
+    find_library(android "android")
+    find_library(glesv2 "GLESv2")
+    find_library(egl "EGL")
+    set(cpufeatures_lib "cpufeatures")
+    set(oboe_lib "oboe")
+    set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+
+    add_compile_definitions(
+            [[JUCE_ANDROID=1]]
+            [[JUCE_DONT_AUTO_OPEN_MIDI_DEVICES_ON_MOBILE=1]]
+            [[JUCE_PUSH_NOTIFICATIONS=1]]
+            [[JUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"]]
+            [[JUCE_USE_ANDROID_OBOE=0]]
+            [[JUCEAAP_HAVE_AUDIO_PLAYHEAD_NEW_POSITION_INFO=1]]
+    )
+
+    target_link_libraries(Os251
+            PRIVATE
+            ${log}
+            ${android}
+            ${glesv2}
+            ${egl}
+    )
+
+endif (ANDROID)
+# <-- end JUCEAAP specific
